osgp/platform/osgp-adapter-ws-shared/src/main/java/org/opensmartgridplatform/adapter/ws/domain/entities/ApplicationKeyLookupKey.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package org.opensmartgridplatform.adapter.ws.domain.entities;
+
+import static org.opensmartgridplatform.shared.utils.StacktraceUtils.currentStacktrace;
+
+import java.io.Serializable;
+import java.util.Objects;
+import javax.persistence.Embeddable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Embeddable
+public final class ApplicationKeyLookupKey implements Serializable {
```

Comment:
This looks like to perform exactly the same function as the `NotificationWebServiceLookupKey`, which I would describe as lookup some application specific (or: organization + application specific) configuration data based on the `organisationIdentification` and `applicationName`.
Wouldn't it be better not to duplicate this identical functionality but to reuse the existing `NotificationWebServiceLookupKey` as embedded ID for the `ApplicationKeyConfiguration`. Of course then the name does not make sense, which looks like it could be resolved by renaming the lookup key class to something like `ApplicationDataLookupKey`, a refactoring that appears more elegant than duplication.

Reply:
That's indeed better. Updated the code as suggested.